components/logins/ffi/src/lib.rs

```diff
@@ -35,6 +35,43 @@ pub unsafe extern "C" fn sync15_passwords_state_new(
     })
 }
 
+unsafe fn bytes_to_key_string(key_bytes: *const u8, len: usize) -> Option<String> {
+    if len == 0 {
```

Comment:
Why are you using `Option` and `assert!` rather than `Result`?

Reply:
Mainly because generally we tend to assert about null pointers rather than returning a Result for them.